Initial setup

YendaRamSurya opened the pull request:
### Module 1 Draft 

Hey Marco !!, 

Following branch contains the info of Module1 which tries to demonstrate a quick tutorial on pytorch vs pytorch lightning differences. 

Overview – PyTorch vs. PyTorch Lightning: Start by comparing raw PyTorch with the high-level Lightning framework. PyTorch offers flexibility but requires you to write your own training loops, handle device placement, logging, etc. – a lot of boilerplate code. PyTorch Lightning builds on PyTorch to automate these engineering details, providing a structured interface for training, validation, and testing. In essence, Lightning “wraps” PyTorch: you still write PyTorch code, but Lightning’s Trainer handles the rest (loops, checkpoints, multi-GPU, etc.) . This lets you focus on research code (the model and training logic) while Lightning handles engineering code (loop boilerplate, logging, hardware acceleration)  . Importantly, Lightning doesn’t abstract away PyTorch’s flexibility – you can always write custom PyTorch in your LightningModule, maintaining full control .

**Key Components of PyTorch Lightning:** This section introduces the Lightning **programming model**, explaining how Lightning organizes code at a high level. The three core components are:

- **LightningModule**
    
- **LightningDataModule:**
    
- **Trainer**

Slides for the Tutorial 1 can be found over here: 

https://docs.google.com/presentation/d/1xxNUHAmSkHztXaisYV99THESMa4bfFCtw255BXWHwRY/edit?usp=sharing

Folder for slides: https://drive.google.com/drive/folders/1bRlrp4DGYJt3m7HW8LXL-kTNM_u-OfOa?usp=sharing

We can incorporate so many changes to this site:

Documentation: https://marcopeix.github.io/pytorch_lightning_course/Introduction/introduction_part_1/

Looking forward for the review!!
Ram Surya yenda

YendaRamSurya committed:
commiting mkdocs and env.yml for the course work

YendaRamSurya committed:
Adding the additional CSS Sheet

YendaRamSurya committed:
pushing code for base pytorch model

YendaRamSurya committed:
pushing lightning code

YendaRamSurya committed:
ignoring the logs

YendaRamSurya committed:
modified intro notebook

YendaRamSurya committed:
fixing the documentation

YendaRamSurya committed:
updating the docs

marcopeix commented:
Great job! Here are my comments:

I notice that the static website does not render correctly the LaTex code. There are also elements in the table of contents that don't work. When clicked, they don't go anywhere.

Otherwise, code looks good! 

Once the issues of the static sites are addressed, I'll merge! Otherwise, we can also drop the static site, as I don't think it's necessary. Notebooks alone are enough.

YendaRamSurya committed:
Updating the mkdocs and css

- updated the css for math formating
- updated the documentation to fit for math formatting

YendaRamSurya commented:
Hey !! Marco, Just updated the static site fix, thank you for the review and feedback. 

https://marcopeix.github.io/pytorch_lightning_course/Introduction/introduction_part_1/

marcopeix commented:
The LaTex equation is still not displayed correctly:
<img width="855" alt="Screenshot 2025-05-14 at 10 33 23 AM" src="https://github.com/user-attachments/assets/d1fb1d41-9c42-4186-a29d-a3bdb1dfd709" />

On the sidebar with the table of contents, every link under "PyTorch Lightning implementation" doesn't work. I think it's because of the "code" because we wrote `DataModule` instead of DataModule (without formatting it as code).

YendaRamSurya committed:
pushing the changes to docs

YendaRamSurya commented:
> The LaTex equation is still not displayed correctly: <img alt="Screenshot 2025-05-14 at 10 33 23 AM" width="855" src="https://private-user-images.githubusercontent.com/19268689/443709219-d1fb1d41-9c42-4186-a29d-a3bdb1dfd709.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDczNjYxMjYsIm5iZiI6MTc0NzM2NTgyNiwicGF0aCI6Ii8xOTI2ODY4OS80NDM3MDkyMTktZDFmYjFkNDEtOWM0Mi00MTg2LWEyOWQtYTNiZGIxZGZkNzA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE2VDAzMjM0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxM2UzOWU5NjJlMTdlOTYyYzQ3NDM3MDc5NGZlMWZkYTA3YjQwYjQ1ZDlkMzVhNDRjYWE3NDNjNjg4NzBhMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OwXuS7k3DOFOWS1c0lKe5RWpNj1EBvf-JxIRNoBQh5o">
> 
> On the sidebar with the table of contents, every link under "PyTorch Lightning implementation" doesn't work. I think it's because of the "code" because we wrote `DataModule` instead of DataModule (without formatting it as code).

@marcopeix , thank you for the suggestions and attention to detail, the following doc issues are recovered, in the latest push
https://marcopeix.github.io/pytorch_lightning_course/Introduction/introduction_part_1/#data-generation-formula

marcopeix reviewed (approved):
Amazing work!
